monai/handlers/mlflow_handler.py

```diff
@@ -174,6 +261,16 @@ def _default_epoch_log(self, engine: Engine) -> None:
         current_epoch = self.global_epoch_transform(engine.state.epoch)
         mlflow.log_metrics(log_dict, step=current_epoch)
 
+        events = engine._event_handlers
+
+        for e in events:
+            for handler, _, _ in engine._event_handlers[e]:
+                if isinstance(handler, ValidationHandler):
+                    evaluator_state = getattr(handler.validator, "state", None)
```

Comment:
We usually attach handlers to `trainer` or `evaluator` separately to track `loss` and `val metrics` during training:
https://github.com/Project-MONAI/tutorials/blob/main/modules/engines/unet_training_dict.py#L125
Here the logic is to get `evaluator` from the `trainer` handler, I feel it's different from our `Statshandler` and `TensorBoardHandler` behavior. But seems `ignite` example has some logic similar in theory?
https://github.com/pytorch/ignite/blob/master/examples/mnist/mnist_with_tensorboard.py#L116
@wyli @vfdev-5 What do you think?

Thanks in advance.